### PR TITLE
Add STS Presigner module with a method to generate a presigned EKS token

### PR DESCRIPTION
**Pull request by mullermp**
Add #1848

Needed to revert because of code generation bug. Added a specific case.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: build_tools/aws-sdk-code-generator/lib/aws-sdk-code-generator/code_builder.rb [hunk 1]
```diff
@@ -57,7 +57,9 @@ def source_files(options = {})
           y.yield("#{prefix}/plugins/apig_endpoint.rb", apig_endpoint_class)
         end
         y.yield("#{prefix}.rb", service_module(prefix))
-        y.yield("#{prefix}/customizations.rb", '')
+        unless prefix == 'aws-sdk-sts'
```
Comment (mullermp) [inline build_tools/aws-sdk-code-generator/lib/aws-sdk-code-generator/code_builder.rb]: I think this change is correct unless you know a better one.

File: gems/aws-sdk-core/lib/aws-sdk-sts/presigner.rb [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      end
+
+      # Returns a presigned url for get_caller_identity.
+      #
+      # @option options [Hash] :headers
+      #   Headers that should be signed and sent along with the request. All
+      #   x-amz-* headers must be present during signing. Other headers are
+      #   optional.
+      #
+      # @option options [Integer] :expires_in (60) Number of seconds before
```
Comment (lwoggardner) [inline gems/aws-sdk-core/lib/aws-sdk-sts/presigner.rb]: Does x-amz-expires actually work for STS?,  the documentation link in Presigner refers to S3 and why 60 as a default?

Comment (mullermp) [reply gems/aws-sdk-core/lib/aws-sdk-sts/presigner.rb]: It appears you are correct. I have overlooked that in the code review. Sorry.. 

Docs: https://docs.aws.amazon.com/STS/latest/APIReference/CommonParameters.html

I can remove this from the PR and update the documentation.

Comment (mullermp) [reply gems/aws-sdk-core/lib/aws-sdk-sts/presigner.rb]: I found this issue: https://github.com/aws/aws-sdk-go/issues/2167 that suggests STS has a 15 minute expiration and will not be changing (last year).

**Comment by lwoggardner** [comment]
This works in us-east-1 but not ap-southeast-2, presumably sts endpoint related.  If I hack the presigner to use 'sts.ap-southeast-2.amazonaws.com' as the host then it works.

Should I open a new issue?

```
Aws> require 'open-uri'
=> true
Aws> region = 'ap-southeast-2'
=> "ap-southeast-2"
Aws> URI.parse(Aws::STS::Presigner.new(client: Aws::STS::Client.new(region: region)).get_caller_identity_presigned_url).open.read
OpenURI::HTTPError: 403 Forbidden
from /usr/share/rvm/rubies/ruby-2.4.0/lib/ruby/2.4.0/open-uri.rb:363:in `open_http'
Aws> region = 'us-east-1'
=> "us-east-1"
Aws> URI.parse(Aws::STS::Presigner.new(client: Aws::STS::Client.new(region: region)).get_caller_identity_presigned_url).open.read
=> "<GetCallerIdentityResponse xmlns=\"https://sts.amazonaws.com/doc/2011-06-15/\">\n  <GetCallerIdentityResult>\n    <Arn>arn:aws:iam::000000000000:user/lwoggardner</Arn>\n    <UserId>AIDAxxxxxxxxxxxxxx</UserId>\n    <Account>000000000000</Account>\n  </GetCallerIdentityResult>\n  <ResponseMetadata>\n    <RequestId>94e6ed4a-19ad-11ea-809b-571ee54a5e88</RequestId>\n  </ResponseMetadata>\n</GetCallerIdentityResponse>\n"
```

**Comment by mullermp** [comment]
@lwoggardner Thanks for the PR. I've submitted comments. The original PR was made before we added the STS regional/legacy option. That was an oversight.

**Pull request merged**